sys/posix/sockets/posix_sockets.c

```diff
@@ -48,11 +48,25 @@
 #include "thread_flags.h"
 #endif
 
+#include "ringbuffer.h"
+
 /* enough to create sockets both with socket() and accept() */
 #define _ACTUAL_SOCKET_POOL_SIZE   (SOCKET_POOL_SIZE + \
                                     (SOCKET_POOL_SIZE * SOCKET_TCP_QUEUE_SIZE))
 #define SOCKET_BLKSIZE             (512)
 
+/**
+ * @brief   Receive buffer, large enough to fit a whole packet with max MTU
+ *
+ * Todo: How to get used MTU size?
+ */
+#define RECV_BUF_SIZE              (1500)
```

Comment:
We could make this configurable during compile time. Beyond that: we have the basis for Path-MTU discovery in place in GNRC (and I believe so does lwIP). The only thing missing is an implementation of the Destination Cache in the NIB that would be the entry point for storage (based on what ICMPv6 errors report) and look-up.

Reply:
Wow ehm ... I agree to the first sentence that we can make it configurable, but I don't know any term you mentioned after that :see_no_evil: As soon such a mechanism is in place I'm happy to replace the compile time configured variable

Reply:
NVM then. Let's first try to find a solution with yet another buffer :-).